mscore/plugin/api/score.h

```diff
@@ -31,39 +31,52 @@ class Measure;
+      /** The duration of the score in seconds (read only) */
+      Q_PROPERTY(int                            duration          READ duration)
       /** The list of the excerpts (linked parts) */
       Q_PROPERTY(QQmlListProperty<Ms::PluginAPI::Excerpt>  excerpts   READ excerpts)
       /** The first measure of the score */
-      Q_PROPERTY(Ms::PluginAPI::Measure*       firstMeasure      READ firstMeasure)
-//       Q_PROPERTY(Ms::Measure*                   firstMeasureMM    READ firstMeasureMM)
-//       Q_PROPERTY(int                            harmonyCount      READ harmonyCount)
-//       Q_PROPERTY(bool                           hasHarmonies      READ hasHarmonies)
-//       Q_PROPERTY(bool                           hasLyrics         READ hasLyrics)
-//       Q_PROPERTY(int                            keysig            READ keysig)
+      Q_PROPERTY(Ms::PluginAPI::Measure*        firstMeasure      READ firstMeasure)
+      Q_PROPERTY(Ms::PluginAPI::Measure*        firstMeasureMM    READ firstMeasureMM)
+      /** Number of harmonoes (chord symbols) in the score */
+      Q_PROPERTY(int                            harmonyCount      READ harmonyCount)
+      /** Whether the score has harmonies (chord symbols) */
+      Q_PROPERTY(bool                           hasHarmonies      READ hasHarmonies)
+      /** Whether the score has lyrics */
+      Q_PROPERTY(bool                           hasLyrics         READ hasLyrics)
+      /// The key signature of the score, in number of accidentals,
+      /// negative for flats, postitive for sharps (read only)
+      Q_PROPERTY(int                            keysig            READ keysig)
       /** The last measure of the score */
-      Q_PROPERTY(Ms::PluginAPI::Measure*       lastMeasure       READ lastMeasure)
-//       Q_PROPERTY(Ms::Measure*                   lastMeasureMM     READ lastMeasureMM)
+      Q_PROPERTY(Ms::PluginAPI::Measure*        lastMeasure       READ lastMeasure)
+      Q_PROPERTY(Ms::PluginAPI::Measure*        lastMeasureMM     READ lastMeasureMM)
       /** The last score segment */
-      Q_PROPERTY(Ms::PluginAPI::Segment*       lastSegment       READ lastSegment) // TODO: make it function? Was property in 2.X, but firstSegment is a function...
-//       Q_PROPERTY(int                            lyricCount        READ lyricCount)
-// //       Q_PROPERTY(QString                        name              READ name           WRITE setName)
+      Q_PROPERTY(Ms::PluginAPI::Segment*        lastSegment       READ lastSegment) // TODO: make it function? Was property in 2.X, but firstSegment is a function...
+      /** Number of lyrics sylables in the score */
+      Q_PROPERTY(int                            lyricCount        READ lyricCount)
+      /** The name of the score, without path leading to it and extension */
+      Q_PROPERTY(QString                        name              READ name           WRITE setName)
```

Comment:
Question is whether we really want to overwrite ScoreElement::name here (which gives "Score" here), or would take another identifiyer for this

Reply:
`fileName` sort of implies the added extension, so I'd opt for `scoreName`